src/solver/interpolate/bwd_trilinear_interpolate.cpp

```diff
@@ -0,0 +1,145 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/activ.hpp"
+#include "miopen/conv_solution.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include "miopen/miopen.h"
+#include <miopen/interpolate/solvers.hpp>
+#include <miopen/interpolate/utils.hpp>
+
+#include <miopen/interpolate/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/interpolate.hpp>
+#include <miopen/target_properties.hpp>
+
+#define LOCAL_SIZE_BWD_TRILINEAR 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace interpolate {
+
+bool IsOverRocmTrilinearBwd(const miopen::interpolate::BwdProblemDescription& problem)
+{
+    TensorDescriptor input_grad_desc  = problem.GetInputGradDesc();
+    TensorDescriptor output_grad_desc = problem.GetOutputGradDesc();
+    auto dtype                        = input_grad_desc.GetType();
+
+    float scale_h =
+        static_cast<float>(output_grad_desc.GetLengths()[2]) / input_grad_desc.GetLengths()[2];
+    float scale_w =
+        static_cast<float>(output_grad_desc.GetLengths()[3]) / input_grad_desc.GetLengths()[3];
+    float scale_d =
+        static_cast<float>(output_grad_desc.GetLengths()[4]) / input_grad_desc.GetLengths()[4];
+
+    if(dtype == miopenHalf || dtype == miopenBFloat16)
+    {
+        if(scale_h + scale_w + scale_d < 3.1f)
+            return false;
+    }
+    else if(dtype == miopenFloat)
+    {
+        if(scale_h + scale_w + scale_d <= 6.0f)
+            return false;
+    }
+
+    return true;
+}
+
+bool InterpolateTrilinearBackward::IsApplicable(
+    const ExecutionContext&, const miopen::interpolate::BwdProblemDescription& problem) const
+{
+    if(problem.GetMode() != miopenInterpolateMode_t::MIOPEN_INTERPOLATE_MODE_TRILINEAR)
+        return false;
+    if(!IsOverRocmTrilinearBwd(problem))
+        return false;
+
+    return true;
+}
+
+ConvSolution InterpolateTrilinearBackward::GetSolution(
+    const ExecutionContext& context,
+    const miopen::interpolate::BwdProblemDescription& problem) const
+{
+    std::ignore = context;
```

Comment:
you can add maybe_unused instead in the params. Ref: https://en.cppreference.com/w/cpp/language/attributes/maybe_unused

Reply:
This type of "ignore params" was accepted in many previous PRs. Maybe I will keep it to avoid making a lot of unnecessary changes in the code. Thank you in advance for the new method.